### PR TITLE
[GLPI 9.5] Impact Analysis - Add link to KnowbaseItem

### PR DESCRIPTION
**Pull request by tsmr**
Ticket https://github.com/glpi-project/glpi/issues/7093

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #7093

### CODE REVIEW

File: inc/impact.class.php [hunk 1]
```diff
@@ -856,20 +856,107 @@ public static function searchAsset(
...
+         //entities restrict
+         // Groups
+         if ($forceall || $has_session_groups) {
+            if (Session::getLoginUserID()) {
+               $restrict = getEntitiesRestrictCriteria('glpi_groups_knowbaseitems', '', '', true, true);
+               $entities_restrict['OR'][] = [
+                  'glpi_groups_knowbaseitems.groups_id' => count($_SESSION["glpigroups"])
+                     ? $_SESSION["glpigroups"]
+                     : [-1],
+                  'OR' => [
```
Comment (trasher) [inline inc/impact.class.php:909]: I guess this extra OR is useless

**Comment by AdrienClairembault** [comment]
Just to make sure I understand correctly : 
- You allowed KnowbaseItem in the impact config
- You want the visiblity restriction of KnowbaseItem to be applied in the impact graph (if the user is not allowed to see the KnowbaseItem then he shouldn't see it or use it on the graph).

Is that correct ?

**Comment by tsmr** [comment]
@AdrienClairembault You're right

**Comment by AdrienClairembault** [comment]
This should be covered by #7268 since it looks like you copied the code from `KnowbaseItem::getVisibilityCriteria()`.

Feel free to reopen if that's not the case and the problem is not fixed.

**Comment by tsmr** [comment]
Sorry but i have SQL errors when i want to link a KnowbaseITem

```
[2020-05-20 07:06:26] glpisqllog.ERROR: DBmysql::query() in C:\wamp64\www\glpi95\inc\dbmysql.class.php line 305
  *** MySQL query error:
  SQL: SELECT COUNT(*) AS total FROM `glpi_knowbaseitems` LEFT JOIN `glpi_knowbaseitems_users` ON (`glpi_knowbaseitems_users`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_knowbaseitems_profiles` ON (`glpi_knowbaseitems_profiles`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) LEFT JOIN `glpi_entities_knowbaseitems` ON (`glpi_entities_knowbaseitems`.`knowbaseitems_id` = `glpi_knowbaseitems`.`id`) WHERE ((LOWER(`glpi_knowbaseitems`.`name`)) LIKE '%%') AND  NOT (`glpi_knowbaseitems`.`id` IN ('-1')) AND (`glpi_knowbaseitems`.`entities_id` IN ('1', '198', '51', '68', '24', '27', '69', '96', '244'))
  Error: Champ 'glpi_knowbaseitems.entities_id' inconnu dans where clause
  Backtrace :
  inc\dbmysqliterator.class.php:95                   
  inc\dbmysql.class.php:849                          DBmysqlIterator->execute()
  inc\impact.class.php:905                           DBmysql->request()
  ajax\impact.php:65                                 Impact::searchAsset()
  {"user":"2@P555","mem_usage":"0.010\", 3.28Mio)"} 
```

**Pull request closed without merging**